lib/ransack/adapters/active_record/context.rb

```diff
@@ -183,13 +189,13 @@ def build_correlated_subquery(association)
           join_root = join_constraints.shift
           join_table = join_root.left
           correlated_key = join_root.right.expr.left
+          if (correlated_key.is_a?(Arel::Nodes::Equality))
+            correlated_key = correlated_key.left
+          end
           subquery = Arel::SelectManager.new(association.base_klass)
           subquery.from(join_root.left)
+          subquery.where(join_root.right.expr)
           subquery.project(correlated_key)
-          join_constraints.each do |j|
-            subquery.join_sources << Arel::Nodes::InnerJoin.new(j.left, j.right)
-          end
-          subquery.where(correlated_key.eq(primary_key))
```

Comment:
This last line was needed to correlate the id on the outer query to the id on the inner query (a kind of join constraint). It looks like this just moved up a few lines and the `correlated_key` is no longer a named variable (`join_root.right.expr` instead), is that right?